ios/FluentUI/Button/FluentButtonStyle.swift

```diff
@@ -0,0 +1,98 @@
+public struct FluentButtonStyle: SwiftUI.ButtonStyle {
+    public init(style: ButtonStyle, size: ButtonSizeCategory) {
+        self.style = style
+        self.size = size
+        self.tokenSet = ButtonTokenSet(style: { style }, size: { size })
+    }
+
+    public func makeBody(configuration: Configuration) -> some View {
+        tokenSet.update(fluentTheme)
+        let isPressed = configuration.isPressed
+        let isDisabled = !isEnabled
+        let isFocused = isFocused
+
+        let cornerRadius = tokenSet[.cornerRadius].float
+        let shadowToken = (isDisabled || isFocused || isPressed) ? ButtonToken.shadowPressed : ButtonToken.shadowRest
+        let shadowInfo = tokenSet[shadowToken].shadowInfo
+
+        let foregroundColor: Color
+        let borderColor: Color
+        let backgroundColor: Color
+        if isDisabled {
+            foregroundColor = Color(tokenSet[.foregroundDisabledColor].uiColor)
+            borderColor = Color(tokenSet[.borderDisabledColor].uiColor)
+            backgroundColor = Color(tokenSet[.backgroundDisabledColor].uiColor)
+        } else if isPressed || isFocused {
+            foregroundColor = Color(tokenSet[.foregroundPressedColor].uiColor)
+            borderColor = Color(tokenSet[.borderPressedColor].uiColor)
+            backgroundColor = Color(tokenSet[.backgroundPressedColor].uiColor)
+        } else {
+            foregroundColor = Color(tokenSet[.foregroundColor].uiColor)
+            borderColor = Color(tokenSet[.borderColor].uiColor)
+            backgroundColor = Color(tokenSet[.backgroundColor].uiColor)
+        }
+
+        @ViewBuilder var backgroundView: some View {
+            if isFloatingStyle {
+                backgroundColor.clipShape(Capsule())
+            } else {
+                backgroundColor.clipShape(RoundedRectangle(cornerRadius: cornerRadius))
+            }
+        }
+
+        @ViewBuilder var overlayView: some View {
+            if borderColor != .clear {
+                if isFloatingStyle {
+                    RoundedRectangle(cornerRadius: cornerRadius, style: .continuous)
+                        .stroke(style: .init(lineWidth: tokenSet[.borderWidth].float))
+                        .foregroundStyle(borderColor)
+                } else {
+                    Capsule()
+                        .stroke(style: .init(lineWidth: tokenSet[.borderWidth].float))
+                        .foregroundStyle(borderColor)
+                }
+            }
+        }
+
+        return configuration.label
+            .font(Font(tokenSet[.titleFont].uiFont))
+            .foregroundStyle(foregroundColor)
+            .padding(edgeInsets)
+            .frame(minHeight: ButtonTokenSet.minContainerHeight(style: style, size: size))
+            .background(backgroundView)
+            .overlay { overlayView }
+            .applyShadow(shadowInfo: shadowInfo)
+            .pointerInteraction(isEnabled)
+    }
+
+    @ObservedObject private var tokenSet: ButtonTokenSet
```

Comment:
Why is this an observed object when it's made by this struct, and not accessed externally?

Reply:
This, along with a lower version, were inherited from an older version of `FluentButton` that tried to synchronize various stateful elements with a wrapping `UIHostingController`. See `MSFAvatar` for an example of that pattern.

Now that this is pure SwiftUI, you're right that `@ObservedObject` is no longer necessary. I'll simplify this.